lib/db_connection.ex

```diff
@@ -793,7 +798,14 @@ defmodule DBConnection do
   `transaction/3` call inside another `transaction/3` will be treated
   the same as `run/3`.
 
-  ### Options
+  > #### Checkout failures {: .warning}
+  >
+  > If we cannot check out a connection from the pool, this function raises a
+  > `DBConnection.ConnectionError` exception. This could also happen in some
+  > other cases, so if you want to handle this case, you should rescue
+  > `DBConnection.ConnectionError` exceptions when using `run/3`.
```

Comment:
"This could also happen in some other cases". This is a bit unclear. This could also happen? This means what here: raising? and which other cases? :)

Reply:
@josevalim it's worse than I thought, because we raise a `DBConnection.ConnectionError` even for bad return values from callbacks. I’m thinking for those it would be better to have a different error, but for now I amended the PR with a bit better docs.

Reply:
bad return sounds like an ArgumentError (kind of the opposite tho) and I think separate exception makes sense as these are always bugs.